File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        max_water=0
        l=0
        r=len(height)-1
        while l<r:
            width=r-l
            h=min(height[l], height[r])
            ar=h*width
            max_water=max(ar,max_water)
            if height[l] > height[r]:
                r-=1
            else:
                l+=1
        return max_water
        

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int max_area = 0;

        while(left <= right && left < height.size()-1)
        {
            max_area = std::max(max_area, std::min(height[left], height[right])*(right-left));
            if(height[left] <= height[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        return max_area;
    }
};